android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -513,6 +713,620 @@ public void onConfigurationChanged(Configuration newConfig) {
+                            TextView newContentButtonText = (TextView) newContentButton.findViewById(R.id.new_content_button_text);
+                            newContentButton.setOnClickListener(new View.OnClickListener() {
+                                @Override
+                                public void onClick(View v) {
+                                    //@TODO alex check why visibility change doesn't work
+                                    newContentButtonText.setVisibility(View.INVISIBLE);
+                                    loadingSpinner.setVisibility(View.VISIBLE);
+
+                                    newContentButton.invalidate();
+
+                                    newContentButton.setClickable(false);
+                                    SharedPreferencesManager.getInstance().writeBoolean(BravePreferenceKeys.BRAVE_NEWS_CHANGE_SOURCE, false);
+                                    if (!isShowNewsOn) {
+                                        isShowNewsOn = true;
+                                    }
+                                    refreshFeed();
+
+                                    newContentButtonText.setVisibility(View.VISIBLE);
+                                    loadingSpinner.setVisibility(View.GONE);
+                                    newContentButton.setClickable(true);
+                                    newContentButton.setVisibility(View.INVISIBLE);
+
+                                    newContentButton.invalidate();
+                                }
+                            });
+                        }
+
+                    } catch (Exception e) {
+                        Log.e("bn",
+                                "Exception  addOnGlobalLayoutListener e: "
+                                        + e);
+                    }
+                }
+            });
+
+        RecyclerView.LayoutManager manager = recyclerView.getLayoutManager();
+        if(manager instanceof LinearLayoutManager) {     
+            LinearLayoutManager linearLayoutManager = (LinearLayoutManager) manager;
+            firstVisiblCard = linearLayoutManager.findFirstVisibleItemPosition();
+
+            recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+                private int lastFirstVisibleItem;
+
+                @Override
+                public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
+                    super.onScrollStateChanged(recyclerView, newState);
+
+                    int firstCompletelyVisibleItemPosition = linearLayoutManager.findFirstCompletelyVisibleItemPosition(); 
+
+                    if (newState == RecyclerView.SCROLL_STATE_DRAGGING) {
+                        endCardViewTime = System.currentTimeMillis();
+                        long timeDiff = endCardViewTime - startCardViewTime;
+                        // if viewed for more than 100 ms send the event
+                        if (timeDiff > 100) {
```

Comment:
where does the number 100 coming from?

Reply:
added it as a static constant in `BraveNewsUtils.java` 
`public static final int BRAVE_NEWS_VIEWD_CARD_TIME = 100; // milliseconds`